airflow/www/gunicorn_config.py

```diff
@@ -37,3 +42,18 @@ def on_starting(server):
 
     # Load providers before forking workers
     ProvidersManager().connection_form_widgets
+
+
+def post_fork(server, worker):
+    """
+    Needed to workoround open-telemetry not working well with gunicorn fork model.
+
+    See more: https://opentelemetry-python.readthedocs.io/en/latest/examples/fork-process-model/README.html
+    """
+    server.log.info("Worker spawned (pid: %s)", worker.pid)
+
+    resource = Resource.create(attributes={"service.name": "airflow-service"})
```

Comment:
here we need to take the service name from `OTEL_SERVICE_NAME` environment variable.